packages/editor/src/components/provider/index.js

```diff
@@ -214,8 +208,12 @@ export const ExperimentalEditorProvider = withRegistryProvider(
 
 		// Sets the right rendering mode when loading the editor.
 		useEffect( () => {
-			setRenderingMode( settings.defaultRenderingMode ?? 'post-only' );
-		}, [ settings.defaultRenderingMode, setRenderingMode ] );
+			setRenderingMode(
+				settings.defaultRenderingMode ?? post.type === 'wp_template'
+					? 'all'
```

Comment:
I'm wondering if there's a need to check the post type here, as `defaultRenderingMode` is already defined as this:
```js
	const defaultRenderingMode = postWithTemplate ? 'template-locked' : 'all';
```

Reply:
I added this as in the post editor `defaultRenderingMode` is undefined by default, which I am guessing was intentional in order to be able to set it to `post-only` in that context, but maybe there is a different way to handle this default setting 🤔

Reply:
It was better to set the default in the post editor - [done here](https://github.com/WordPress/gutenberg/pull/57700/commits/1d923732970a2d52a38ef6c92b371f7265c35e48)